components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/selection/GeckoSelectionActionDelegate.kt

```diff
@@ -19,25 +19,30 @@ import org.mozilla.geckoview.BasicSelectionActionDelegate
 open class GeckoSelectionActionDelegate(
     activity: Activity,
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal val customDelegate: SelectionActionDelegate
+    internal val customDelegate: SelectionActionDelegate,
+    private val actionSorter: ((Array<String>) -> Array<String>)? = null
 ) : BasicSelectionActionDelegate(activity) {
 
     companion object {
         /**
          * @returns a [GeckoSelectionActionDelegate] if [customDelegate] is non-null and [context]
          * is an instance of [Activity]. Otherwise, returns null.
          */
-        fun maybeCreate(context: Context, customDelegate: SelectionActionDelegate?): GeckoSelectionActionDelegate? {
+        fun maybeCreate(context: Context, customDelegate: SelectionActionDelegate?, actionSorter: ((Array<String>) -> Array<String>)? = null): GeckoSelectionActionDelegate? {
             return if (context is Activity && customDelegate != null) {
-                GeckoSelectionActionDelegate(context, customDelegate)
+                GeckoSelectionActionDelegate(context, customDelegate, actionSorter)
             } else {
                 null
             }
         }
     }
 
     override fun getAllActions(): Array<String> {
-        return super.getAllActions() + customDelegate.getAllActions()
+        return if (actionSorter != null) {
+            actionSorter.invoke(super.getAllActions() + customDelegate.getAllActions())
+        } else {
+            super.getAllActions() + customDelegate.getAllActions()
+        }
```

Comment:
nit: I think we could ask for a `(String) -> R?` mapping, so that we can pass it directly to `sortedBy` here?